plugin.rb

```diff
@@ -567,7 +567,7 @@ def self.denormalized_columns
 
   # Reimplement a couple ActiveRecord methods, but use PluginStore for storage instead
   class DataExplorer::Query
-    attr_accessor :id, :name, :description, :sql
+    attr_accessor :id, :name, :description, :sql, :created_by, :created_at, :username
```

Comment:
I don't think we need to save both `id` and `username`. We should just save `id` and user should be fetched in serializer to pass on to client.

Reply:
What I meant was do not save `username` at all in the row since we are already saving user id and fetch the user details in serializer and pass it on to client.

So removing `username` from here means removing it from everywhere you added it and fetching user in serializer instead.

Note that this (fetching user in serializer) is already needed in the next step which is to show user avatar on index page.

Reply:
Oh, I understand now. That makes a lot of sense, I shouldn't have left the extra variable there.

By fetching user details in serializer, do mean something like this and using `{{query.fetch_user.username}}`? 

```
class DataExplorer::QuerySerializer < ActiveModel::Serializer
    attributes :id, :sql, :name, :description, :param_info, :created_by, :created_at

    def param_info
      object.params.map(&:to_hash) rescue nil
    end

    def fetch_user
      object.user = current_user
    end
  end
```
This does not seem to work, is this the correct way to fetch user?

Reply:
By fetching user details I mean retrieving the user row in serializer.

To get this PR merged the simplest thing we can do right now with minimal changes is:

```ruby
class DataExplorer::QuerySerializer < ActiveModel::Serializer
  attributes :id, :sql, :name, :description, :param_info, :created_by, :created_at, :username

  def param_info
    object.params.map(&:to_hash) rescue nil
  end

  def username
    User.find(created_by).username
  end
end
```

But note that this causes n+1 queries so we need to think of better solution in the next round of changes.

Reply:
```ruby
def username
    User.find(created_by).username
end
```
This just didn't work for me, I tried very hard to make it work. I could not get rid of some errors when I used that^

Can you check the current changes? 
I've simply added  `query.username = User.find(query.created_by).username` in def create() and this works as expected for now. Looks okay?

Reply:
Here is what I meant: https://github.com/discourse/discourse-data-explorer/pull/13/commits/a923fc8af48277bb2f4d92bf28959f3be1d893dc

Can you verify the final changes locally? Also we need to rebase this branch with master.

Once all is done feel free to merge at will. 👍